File: back/src/main/java/com/openclassrooms/mddapi/entities/Article.java
package com.openclassrooms.mddapi.entities;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
public class Article {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    @Lob
    private String content;
    private LocalDateTime createdAt;

    @ManyToOne
    @JoinColumn(name = "author_id")
    private User author;

    @ManyToOne
    @JoinColumn(name = "theme_id")
    private Theme theme;

    @OneToMany(mappedBy = "article")
    private List<Comment> comments;

    @OneToMany(mappedBy = "article")
    private List<Subscription> subscriptions;

    // Getters and setters
}

File: front/src/app/models/article.ts
import { Author } from "./author";
import { Theme } from "./themes";
import { User } from "./user";

export interface Article {

    id : number;
    title: string;
    content: string;
    createdAt: String | null;
    author: Author ;
    theme: Theme | null;
    comments: Comment[] | null;
  }